services/multitenant/tenantstats.go

```diff
@@ -58,19 +68,59 @@ func NewStats(routerDB jobsdb.MultiTenantJobsDB) *MultitenantStatsT {
 	multitenantStat.routerInputRates["batch_router"] = make(map[string]map[string]misc.MovingAverage)
 	multitenantStat.lastDrainedTimestamps = make(map[string]map[string]time.Time)
 	multitenantStat.failureRate = make(map[string]map[string]misc.MovingAverage)
-	multitenantStat.routerTenantLatencyStat = make(map[string]map[string]misc.MovingAverage)
-	multitenantStat.processorStageTime = time.Now()
 	pileUpStatMap := make(map[string]map[string]int)
-	routerDB.GetPileUpCounts(pileUpStatMap)
-	for workspace := range pileUpStatMap {
-		for destType := range pileUpStatMap[workspace] {
-			multitenantStat.AddToInMemoryCount(workspace, destType, pileUpStatMap[workspace][destType], "router")
+	for routerType := range routerDBs {
+		routerDBs[routerType].GetPileUpCounts(pileUpStatMap)
+		for workspace := range pileUpStatMap {
+			for destType := range pileUpStatMap[workspace] {
+				multitenantStat.AddToInMemoryCount(workspace, destType, pileUpStatMap[workspace][destType], routerType)
+			}
 		}
 	}
 
+	multitenantStat.routerTenantLatencyStat = make(map[string]map[string]misc.MovingAverage)
+
+	multitenantStat.processorStageTime = time.Now()
+
+	rruntime.Go(func() {
+		multitenantStat.sendStats()
+	})
+
 	return &multitenantStat
 }
 
+func (multitenantStat *MultitenantStatsT) sendStats() {
+	multiTenantStatTicker := time.NewTicker(multiTenantStatFrequency)
+	defer multiTenantStatTicker.Stop()
+	for range multiTenantStatTicker.C {
+		func() {
+			multitenantStat.routerJobCountMutex.RLock()
+			defer multitenantStat.routerJobCountMutex.RUnlock()
+
+			for tableType, tableStats := range multitenantStat.routerNonTerminalCounts {
+				for workspace, workspaceStats := range tableStats {
+					for destType, destTypeCount := range workspaceStats {
+						pendingEventCountStat := stats.NewTaggedStat("pending_event_count",
+							stats.GaugeType, stats.Tags{
+								"workspace":   workspace,
+								"destType":    destType,
+								"tablePrefix": tableType,
+							})
+						pendingEventCountStat.Gauge(destTypeCount)
+					}
+				}
+			}
+		}()
+	}
```

Comment:
Could we decouple stats keeping functionality from sending metrics to a different component as we discussed?

Reply:
put it in processor